src/components/SearchBar.jsx

```diff
@@ -49,14 +49,15 @@ font-size: 1.5rem;
 let searchId;
 
 export default function SearchBar() {
-  const { setPage } = useContext(TopContext);
+  const { setPage, setPageId } = useContext(TopContext);
   const [search, setSearch] = useState('');
   const [artistsArr, setArtistsArr] = useState([]);
   const [eventsArr, setEventsArr] = useState([]);
   const [noResults, setNoResults] = useState(false);
 
   function handleClick(e, id) {
-    setPage(e.target.name, id);
+    setPage(e.target.attributes.name.nodeValue);
+    setPageId(id);
```

Comment:
It should maybe only change the id if the id is not undefined. But maybe it works fine regardless.